smarts/core/tests/test_env_frame_rate.py

```diff
@@ -0,0 +1,112 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import gym
+import pytest
+
+import logging
+
+import gym
+from smarts.core.utils.frame_monitor import FrameMonitor
+from smarts.core.agent import Agent, AgentSpec
+from smarts.core.agent_interface import AgentInterface, AgentType
+from smarts.core.sensors import Observation
+from smarts.core.utils.episodes import episodes
+
+logging.basicConfig(level=logging.INFO)
+
+AGENT_ID = "Agent-007"
+
+
+@pytest.fixture
+def scenarios():
+    return ["scenarios/tests/multi_agents_loop"]
+
+
+@pytest.fixture
+def seed():
+    return 42
+
+
+class ChaseViaPointsAgent(Agent):
+    def act(self, obs: Observation):
+        if (
+            len(obs.via_data.near_via_points) < 1
+            or obs.ego_vehicle_state.edge_id != obs.via_data.near_via_points[0].edge_id
+        ):
+            return (obs.waypoint_paths[0][0].speed_limit, 0)
+
+        nearest = obs.via_data.near_via_points[0]
+        if nearest.lane_index == obs.ego_vehicle_state.lane_index:
+            return (nearest.required_speed, 0)
+
+        return (
+            nearest.required_speed,
+            1 if nearest.lane_index > obs.ego_vehicle_state.lane_index else -1,
+        )
+
+
+def env_and_spec(scenarios, seed, headless=True, max_episode_steps=None):
+    agent_spec = AgentSpec(
+        interface=AgentInterface.from_type(
+            AgentType.LanerWithSpeed, max_episode_steps=max_episode_steps
+        ),
+        agent_builder=ChaseViaPointsAgent,
+    )
+
+    env = gym.make(
+        "smarts.env:hiway-v0",
+        scenarios=scenarios,
+        agent_specs={AGENT_ID: agent_spec},
+        sim_name=None,
+        headless=headless,
+        visdom=False,
+        timestep_sec=0.1,
+        sumo_headless=True,
+        seed=seed,
+    )
+    return env, agent_spec
+
+
+def test_env_frame_test(scenarios, seed):
+    env, agent_spec = env_and_spec(scenarios, seed)
+    runtime_warnings = 0
+    for episode in episodes(n=10):
+        agent = agent_spec.build_agent()
+        observations = env.reset()
+        episode.record_scenario(env.scenario_log)
+
+        dones = {"__all__": False}
+        while not dones["__all__"]:
+            agent_obs = observations[AGENT_ID]
+            agent_action = agent.act(agent_obs)
+            try:
+                with FrameMonitor(40):
+                    observations, rewards, dones, infos = env.step(
+                        {AGENT_ID: agent_action}
+                    )
+            except Exception as e:
+                runtime_warnings += 1
+                pass
+            episode.record_step(observations, rewards, dones, infos)
+    env.close()
+    assert runtime_warnings > 0
```

Comment:
I will add the logging part of each part of simulation step in another PR since that is a different issue. As for the log of each delta-time,  the minimum, average, and maximum step lengths, I will add them  into the logs of the test for the time being. as you mentioned above.